### PR TITLE
IDENTITY-3229

### PR DESCRIPTION
**Pull request by wattale**

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 1]
```diff
@@ -292,13 +294,40 @@ private void updateBasicApplicationData(int applicationId, String applicationNam
         }
 
         // only if the application has been renamed
-        if (!applicationName.equalsIgnoreCase(storedAppName)) {
+        if (!applicationName.equals(storedAppName)) {
```
Comment (hpmtissera) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: use StringUtils for string comparision

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 2]
```diff
@@ -240,11 +240,15 @@ public void updateApplication(ServiceProvider serviceProvider)
             }
 
             ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
+            String storedAppName = appDAO.getApplicationName(serviceProvider.getApplicationID());
             appDAO.updateApplication(serviceProvider);
             ApplicationPermission[] permissions =
                     serviceProvider.getPermissionAndRoleConfig()
                             .getPermissions();
-            if (permissions != null && permissions.length > 0) {
+            if(!storedAppName.equals(serviceProvider.getApplicationName())){
```
Comment (GayanM) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: check the possibility of having null for storedAppName

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 1 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed with the PR : https://github.com/wso2/carbon-identity/pull/830

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 2 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/830

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 3]
```diff
@@ -272,7 +303,7 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
                 tenantGovReg.delete(applicationNode);
             }
 
-            if (permissions == null) {
+            if (permissions == null || permissions.length == 0) {
```
Comment (prabathabey) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Use String.isEmpty()

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 4]
```diff
@@ -292,13 +294,40 @@ private void updateBasicApplicationData(int applicationId, String applicationNam
...
                 log.debug("Renaming application role from " + storedAppName + " to "
                         + applicationName);
             }
+            PreparedStatement readPermissions = null;
+            ResultSet resultSet = null;
+            try {
+                readPermissions = connection.prepareStatement(ApplicationMgtDBQueries.LOAD_UM_PERMISSIONS);
+                readPermissions.setString(1, "%" + ApplicationMgtUtil.getApplicationPermissionPath() + "%");
+                resultSet = readPermissions.executeQuery();
+                while (resultSet.next()) {
```
Comment (prabathabey) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Break this logic to two separate methods, which would make the code more readable.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 5]
```diff
@@ -2248,6 +2277,75 @@ public void deletePermissionAndRoleConfiguration(int applicationID, Connection c
...
+     * @param applicationName
+     * @param permissions
+     * @throws IdentityApplicationManagementException
+     * @throws SQLException
+     */
+    public void deleteAssignedPermissions(Connection connection, String applicationName, ApplicationPermission[] permissions)
+            throws IdentityApplicationManagementException, SQLException {
+        List<ApplicationPermission> loadPermissions = ApplicationMgtUtil.loadPermissions(applicationName);
+        List<ApplicationPermission> removedPermissions = null;
+        if (loadPermissions != null && loadPermissions.size() > 0) {
```
Comment (prabathabey) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Use list.isEmpty()

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 6]
```diff
@@ -2248,6 +2277,75 @@ public void deletePermissionAndRoleConfiguration(int applicationID, Connection c
...
+            if (permissions == null || permissions.length == 0) {
```
Comment (prabathabey) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Use String.isEmpty()

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 7]
```diff
@@ -2248,6 +2277,75 @@ public void deletePermissionAndRoleConfiguration(int applicationID, Connection c
...
+            for (ApplicationPermission applicationPermission : removedPermissions) {
+                String permissionValue = ApplicationMgtUtil.PATH_CONSTANT +
+                        ApplicationMgtUtil.getApplicationPermissionPath() +
+                        ApplicationMgtUtil.PATH_CONSTANT +
+                        applicationName + ApplicationMgtUtil.PATH_CONSTANT +
+                        applicationPermission.getValue();
+                PreparedStatement selectQuery = null;
+                ResultSet resultSet = null;
+                try {
+                    selectQuery = connection.prepareStatement(ApplicationMgtDBQueries.LOAD_UM_PERMISSIONS_W);
```
Comment (prabathabey) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Break this logic into multiple methods to improve readability.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 3 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/937

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 5 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/937

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 6 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/937

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 7 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/964

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java (hunk 4 above)
Comment (wattale) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java]: Fixed in the PR : https://github.com/wso2/carbon-identity/pull/964

**Pull request merged**